### DISCUSSION TITLE
EJBCA 7.11.0 - Multiple notifications intermittently send when creating, renewing end entity / dynamic variables not displaying value

### DISCUSSION THREAD

**Discussion by kfox13** [Keyfactor/ejbca-ce#369] [Q&A]
Hi,
We've recently upgraded our ejbca instance to 7.11.0 and since doing so are noticing an issue that happens very intermittently (about once in every 10 times).

Occasionally when creating / renewing end entity status,
- End entity recieves the notification we have configured twice (from same action)
- In one of the notifications, the ${requestAdmin.CN} variable shows as the variable name in the recieved email (and not what the vaue of it is)

Do you know why this might be happening? / How to best troubleshoot?

I have had a look in the server log at the time I managed to replicate the issue, but there doesn't seem to be anything relevent in explaing why this intermittently happens.

Thanks

**Comment by primetomas** [comment]
Enable debug logging when it happens, debug logs should tell details of every notification passed through the code.

**Comment by Marc-bo** [comment]
Hi kfox13,
any update on this small issue ?
I got a very similar issue since the upgrade to 7.11, for some End Entity Profile I got spurious notification with empty data... (and no pb with some other with same notification setting).

Marc.

**Comment by kfox13** [comment]
Hi,
Unfortunately not yet sorry... the issue is very intermittent for us so has been deemed low priority. I will update once have managed to look. Thanks

**Comment by primetomas** [comment]
You should get entries in the log when mail fails to be sent. EJBCA uses SMTP and if it can't send with SMTP errors are logged (connection refused or something like that). If it is passed on to the SMTP server EJBCA assumes it is handled.

**Comment by kfox13** [comment]
Hey,
Managed to get some time on this. The issue seems to occur when STATUSINITIALISED option is selected for notifications on the end entity.
Noticed that option shouldn't do anything anymore as per below guide, but it does seem to cause the sending of a notification on the cert being generated.
(https://doc.primekey.com/ejbca/ejbca-operations/ejbca-ca-concept-guide/end-entities-overview/end-entity-profiles-overview/e-mail-notifications)
After clearing it, I can no longer replicate the issue.
The debug log didn't show much other than the notification being sent on collecting the cert.
Thanks
